Sources/Helpers/OAFavoritesHelper.h

```diff
@@ -7,12 +7,30 @@
 //
 
 #import <Foundation/Foundation.h>
+#include <OsmAndCore/IFavoriteLocation.h>
 
-@class OAFavoriteItem;
+@class OAFavoriteItem, OAFavoriteGroup;
 
 @interface OAFavoritesHelper : NSObject
 
 + (NSArray<OAFavoriteItem *> *) getFavoriteItems;
 + (NSArray<OAFavoriteItem *> *) getVisibleFavoriteItems;
++ (NSArray<OAFavoriteGroup *> *) getGroupedFavorites:(QList< std::shared_ptr<OsmAnd::IFavoriteLocation> >)allFavorites;
+
+@end
+
+@interface OAFavoriteGroup : NSObject
+
+@property (nonatomic) NSString* name;
+@property (nonatomic) UIColor* color;
+@property (nonatomic) BOOL isHidden;
+@property (nonatomic) NSMutableArray<OAFavoriteItem*> *points;
+
+- (instancetype) initWithName:(NSString *)name isHidden:(BOOL)isHidden color:(UIColor *)color;
+- (instancetype) initWithPoints:(NSArray<OAFavoriteItem *> *)points name:(NSString *)name isHidden:(BOOL)isHidden color:(UIColor *)color;
+- (NSArray<OAFavoriteItem*> *) getPoints;
```

Comment:
Absolutely no need for getters in Obj c if you have a property (do not port getters from Java).